internal/gengapic/mixins.go

```diff
@@ -56,65 +119,102 @@ func (g *generator) getMixinFiles() []*descriptor.FileDescriptorProto {
 	return files
 }
 
+// getMixinMethods is a convenience method to collect the method descriptors of
+// those methods to be generated based on if they should be included or not.
 func (g *generator) getMixinMethods() []*descriptor.MethodDescriptorProto {
 	methods := []*descriptor.MethodDescriptorProto{}
 	if g.hasLocationMixin() {
-		methods = append(methods, getLocationsMethods()...)
+		methods = append(methods, g.mixins["google.cloud.location.Locations"]...)
 	}
 	if g.hasIAMPolicyMixin() {
-		methods = append(methods, getIAMPolicyMethods()...)
+		methods = append(methods, g.mixins["google.iam.v1.IAMPolicy"]...)
 	}
 	if g.hasLROMixin() {
-		methods = append(methods, getOperationsMethods()...)
+		methods = append(methods, g.mixins["google.longrunning.Operations"]...)
 	}
```

Comment:
IIUC, this can be cleaned up somewhat.
```
for _, mixin_methods := range g.mixins {
    methods = append(methods, mixin_methods)
}
```

Reply:
Almost. Some of the `has` methods have logic in there to prevent the mixin from being generated in the event that 1. the API already overrides some mixin methods 2. the Service-to-generate is the mixin service itself (e.g. generating a client for Operations). 

I could look at pushing that logic into the initialization section though. I will take a look.

Reply:
Good to know about the `has` methods. Don't worry about it if it's a hassle; I just saw a potentially easy transform and thought "why not?"

Reply:
It's a good idea. I'm going to punt on this for now, but I have an idea on how it could be improved. Just requires some plumbing and I'd like to separate it out.